Lib/glyphsLib/builder/user_data.py

```diff
@@ -59,7 +61,18 @@ def to_ufo_master_user_data(self, ufo, master):
 def to_ufo_glyph_user_data(self, ufo, glyph):
     key = GLYPH_USER_DATA_KEY + "." + glyph.name
     if glyph.userData:
-        ufo.lib[key] = dict(glyph.userData)
+        # Convert MATH userData to top-level keys and group them under the same
+        # key so that they are in a more usable/compact form.
+        if GLYPHS_MATH_EXTENDED_SHAPE_KEY in glyph.userData:
+            ufo.lib.setdefault(GLYPHS_MATH_EXTENDED_SHAPE_KEY, []).append(glyph.name)
+            del glyph.userData[GLYPHS_MATH_EXTENDED_SHAPE_KEY]
```

Comment:
why do you need to modify the Glyphs glyph object's userData when converting this to UFO lib?

Reply:
So that it does not get written again by the code below. I’ll figure a way that does not modify the input font.